### ISSUE TITLE
Do you plan to support mbedTLS?

### ISSUE THREAD

**Issue by zhaojh329** [deleisha/evt-tls#11]
Do you plan to support mbedTLS?

**Comment by bl4ck5un** [comment]
Same question!

**Comment by deleisha** [comment]
Yes, we intend to support. Contributors welcome

**Comment by ToiletCommander** [comment]
Right. mbedtls is much easier to build compare to OpenSSL, and it is great to see a lib that supports SSL with libuv with embedtls. Looking forward to the new implemented embedtls version of evt-tls!

**Comment by deleisha** [comment]
Experimentation on mbedtls started, Either I am not fluent yet on it or their API are complete yet. 
Using mbedtls as state black box is little difficult.

**Comment by ToiletCommander** [comment]
[mbedTLS-Tutorial](https://tls.mbed.org/kb/how-to/mbedtls-tutorial)
This link could probably help. It describes how to create TLS connected Sockets using it.  
Good Luck  : )

**Comment by ToiletCommander** [comment]
I just read a bit about the tutorial. embedTLS is hard to implement into evt-tls.
It replaces the original read() function in UNIX and Windows API. Rather than using read(), it let you use its own function.
I think that implementing embedTLS support could be compicated and time-consuming since we might need to modify the implemention inside libuv.

**Comment by ToiletCommander** [comment]
As [LibUV-mbedtls](https://github.com/geminiwen/libuv-mbedtls) do, he re-wrote bio.c from mbedtls and then wrap mbedtls over libuv. I think this could be a good solution for us.

**Comment by deleisha** [comment]
Thanks @ToiletCommander for your effort. 
Perhaps, we don't want to go wolfssl way for mbedtls as that will involved maintaining the `bio.c`. I did a feature request to mbedtls. They leave network IO(nio) handling to the user and don't do buffering.
So, we need to come up with NIO part. Now we know what need to be done. 
I am working on now to have a unified error handling.

**Comment by ToiletCommander** [comment]
Just one question, as you requested the feature, when is the feature going to be implmented?
Because I am uncertain about the version of the mbedtls lib to download.

**Comment by deleisha** [comment]
I don't think it will be in the near future. 
We can go ahead without that as well.
Please get the latest release version because that's the version that I am planning to checked in and use.
